### PR TITLE
avoid ingress_conf_file recreation at each terraform apply

### PR DESCRIPTION
**Pull request by alouvion-aneo**
[infos](https://github.com/aneoconsulting/ArmoniK/issues/946)

=> after this fix : nomore ingress update need detected during the plan

### CODE REVIEW

File: infrastructure/quick-deploy/aws/all/armonik.tf [hunk 1]
```diff
@@ -6,6 +6,20 @@ module "armonik" {
...
+  seq_secret_name                        = kubernetes_secret.seq.id != null ? kubernetes_secret.seq.metadata[0].name : kubernetes_secret.seq.metadata[0].name
+  shared_storage_secret_name             = kubernetes_secret.shared_storage.id != null ? kubernetes_secret.shared_storage.metadata[0].name : kubernetes_secret.shared_storage.metadata[0].name
+  deployed_object_storage_secret_name    = kubernetes_secret.deployed_object_storage.id != null ? kubernetes_secret.deployed_object_storage.metadata[0].name : kubernetes_secret.deployed_object_storage.metadata[0].name
+  deployed_table_storage_secret_name     = kubernetes_secret.deployed_table_storage.id != null ? kubernetes_secret.deployed_table_storage.metadata[0].name : kubernetes_secret.deployed_table_storage.metadata[0].name
+  deployed_queue_storage_secret_name     = kubernetes_secret.deployed_queue_storage.id != null ? kubernetes_secret.deployed_queue_storage.metadata[0].name : kubernetes_secret.deployed_queue_storage.metadata[0].name
+  s3_secret_name                         = length(module.s3_os) > 0 && kubernetes_secret.s3.0.id != null ? kubernetes_secret.s3[0].metadata[0].name : var.s3_secret_name
```
Comment (lemaitre-aneo) [inline infrastructure/quick-deploy/aws/all/armonik.tf:21]: You should not need the ternary trick as you refer directly to the name

File: infrastructure/quick-deploy/aws/all/armonik.tf [hunk 2]
```diff
@@ -18,7 +18,7 @@ module "armonik" {
   deployed_object_storage_secret_name    = kubernetes_secret.deployed_object_storage.id != null ? kubernetes_secret.deployed_object_storage.metadata[0].name : kubernetes_secret.deployed_object_storage.metadata[0].name
   deployed_table_storage_secret_name     = kubernetes_secret.deployed_table_storage.id != null ? kubernetes_secret.deployed_table_storage.metadata[0].name : kubernetes_secret.deployed_table_storage.metadata[0].name
   deployed_queue_storage_secret_name     = kubernetes_secret.deployed_queue_storage.id != null ? kubernetes_secret.deployed_queue_storage.metadata[0].name : kubernetes_secret.deployed_queue_storage.metadata[0].name
-  s3_secret_name                         = length(module.s3_os) > 0 && kubernetes_secret.s3.0.id != null ? kubernetes_secret.s3[0].metadata[0].name : var.s3_secret_name
+  s3_secret_name                         = try(kubernetes_secret.s3.0.id, null) != null ? kubernetes_secret.s3[0].metadata[0].name : var.s3_secret_name
```
Comment (lemaitre-aneo) [inline infrastructure/quick-deploy/aws/all/armonik.tf:21]: you can just write:
```suggestion
  s3_secret_name                         = can(coalesce(kubernetes_secret.s3.0.id)) ? kubernetes_secret.s3[0].metadata[0].name : var.s3_secret_name
```

**Review by alouvion-aneo — COMMENTED** [review]
I did update in allinone deployment I need to report in the normal deployment 
=> var for secret resource names

File: infrastructure/quick-deploy/aws/all/variables.tf [hunk 3]
```diff
@@ -677,3 +677,70 @@ variable "image_tags" {
...
+  description = "the name of the deployed-queue-storage secret"
+  type        = string
+  default     = "deployed-queue-storage"
+}
+
+variable "s3_secret_name" {
+  description = "the name of the S3 secret"
+  type        = string
+  default     = "s3"
+}
```
Comment (lemaitre-aneo) [inline infrastructure/quick-deploy/aws/all/variables.tf:746]: There is no need to variabilize this at the root module level

Comment (alouvion-aneo) [reply infrastructure/quick-deploy/aws/all/variables.tf:746]: indeed I was thinking the same and Lilia will be happy :)

File: infrastructure/quick-deploy/aws/all/armonik.tf [hunk 4]
```diff
@@ -6,6 +6,20 @@ module "armonik" {
...
+  seq_secret_name                        = kubernetes_secret.seq.id != null ? kubernetes_secret.seq.metadata[0].name : kubernetes_secret.seq.metadata[0].name
+  shared_storage_secret_name             = kubernetes_secret.shared_storage.id != null ? kubernetes_secret.shared_storage.metadata[0].name : kubernetes_secret.shared_storage.metadata[0].name
+  deployed_object_storage_secret_name    = kubernetes_secret.deployed_object_storage.id != null ? kubernetes_secret.deployed_object_storage.metadata[0].name : kubernetes_secret.deployed_object_storage.metadata[0].name
+  deployed_table_storage_secret_name     = kubernetes_secret.deployed_table_storage.id != null ? kubernetes_secret.deployed_table_storage.metadata[0].name : kubernetes_secret.deployed_table_storage.metadata[0].name
+  deployed_queue_storage_secret_name     = kubernetes_secret.deployed_queue_storage.id != null ? kubernetes_secret.deployed_queue_storage.metadata[0].name : kubernetes_secret.deployed_queue_storage.metadata[0].name
+  s3_secret_name                         = can(coalesce(kubernetes_secret.s3.0.id)) ? kubernetes_secret.s3[0].metadata[0].name : var.s3_secret_name
```
Comment (lemaitre-aneo) [inline infrastructure/quick-deploy/aws/all/armonik.tf:21]:
```suggestion
  s3_secret_name                         = can(coalesce(kubernetes_secret.s3.0.id)) ? kubernetes_secret.s3[0].metadata[0].name : ""
```

**Review by lemaitre-aneo — APPROVED** [review]

**Pull request merged**